src/main/java/com/minecolonies/colony/Field.java

```diff
@@ -0,0 +1,467 @@
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Handles the field class.
+ */
+public class Field extends Container
+{
+    /**
+     * Tag to store the location.
+     */
+    private static final String TAG_LOCATION = "location";
+
+    /**
+     * Tag to store if the field has been taken.
+     */
+    private static final String TAG_TAKEN = "taken";
+
+    /**
+     * Tag to store the fields positive length.
+     */
+    private static final String TAG_LENGTH_PLUS = "length+";
+
+    /**
+     * Tag to store the fields positive width.
+     */
+    private static final String TAG_WIDTH_PLUS = "width+";
+
+    /**
+     * Tag to store the fields negative length.
+     */
+    private static final String TAG_LENGTH_MINUS = "length-";
+
+    /**
+     * Tag to store the fields negative width.
+     */
+    private static final String TAG_WIDTH_MINUS = "width-";
+
+    /**
+     * Tag to store the fields stage.
+     */
+    private static final String TAG_STAGE = "stage";
+
+    /**
+     * The max width/length of a field.
+     */
+    private static final int MAX_RANGE = 5;
+
+    /**
+     * The fields location.
+     */
+    private BlockPos location;
+
+    /**
+     * The colony of the field.
+     */
+    private final Colony colony;
+
+    /**
+     * Has the field be taken by any worker?
+     */
+    private boolean taken = false;
+
+    /**
+     * Checks if the field needsWork (Hoeig, Seedings, Farming etc)
+     */
+    private boolean needsWork = false;
+
+    /**
+     * Has the field been planted?
+     */
+    private FieldStage fieldStage = FieldStage.EMPTY;
+
+    /**
+     * The set seed type for the field.
+     */
+    private ItemSeeds seed;
+
+    /**
+     * The length to plus x of the field.
+     */
+    private int lengthPlusX;
+
+    /**
+     * The width to plus z of the seed.
+     */
+    private int widthPlusZ;
+
+    /**
+     * The length to minus xof the field.
+     */
+    private int lengthMinusX;
+
+    /**
+     * The width to minus z of the seed.
+     */
+    private int widthMinusZ;
+
+    /**
+     * The inventorySlot of the field.
+     */
+    private InventoryField inventory;
+
+    /**
+     * Describes the stage the field is in.
+     * Like if it has been hoed, planted or is empty.
+     */
+    public enum FieldStage
+    {
+        EMPTY,
+        HOED,
+        PLANTED
+    }
+
+
+     /**
+     * Private constructor to create field from NBT.
+     * @param colony the colony the field belongs to.
+     */
+    private Field(Colony colony)
+    {
+       this.colony = colony;
+    }
+
+    /**
+     * Creates an instance of our field container, this may be serve to open the GUI.
+     * @param inventory the field inventory.
+     * @param playerInventory the player inventory.
+     * @param world the world.
+     * @param location the position of the field.
+     */
+    public Field(InventoryField inventory, InventoryPlayer playerInventory, World world, BlockPos location)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) It is a good practice to call super() in a constructor [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ACallSuperInConstructor)